index.bs

```diff
@@ -3631,6 +3634,377 @@ opened</dfn> steps given |window|, |type| and |message|.
+    url: text,
+    method: text,
+    headers: [ *NetworkHeader ],
+    cookies: [ *NetworkCookie ],
+    headersSize: uint,
+    bodySize: uint,
+};
+</pre>
+
+The <code>RequestData</code> type represents an ongoing network request.
+
+<div algorithm>
+
+To <dfn>get the request data</div> given |request|:
+
+1. Let |url| be the result of running the [=URL serializer=] with |request|'s
+   [=URL=].
+
+1. Let |method| be |request|'s [=method=].
+
+1. let |body size| be 0.
+
+1. Let |body| be request's [=body=].
+
+1. If |body| is a [=byte sequence=], let |body size| be the length of that
+   sequence. Otherwise if |body| is a [=body=] then let |body size| be that
+   body's [=length=] if it's not null, or 0 otherwise.
+
+   TODO: Should we handle the case where the body length is not yet known?
+
+1. Let |headers size| be the size in bytes of |request|'s [=headers list=] when
+   serialized as mandated by [=HTTP=].
+
+   Note: for protocols which allow header compression, this is the compressed
+   size of the headers, as sent over the network.
+
+1. Let |headers| be an empty list.
+
+1. Let |cookies| be an empty list.
+
+1. For each (|name|, |value|) in |request|'s [=headers list=]:
+
+  1. Append the result of [=get a Header=] with |name| and |value| to |headers|.
+
+  1. If |name| is a [=byte-case-insensitive=] match for "<code>Cookie</code>" then:
+
+    1. TODO: Add cookies
+
+    CDP puts all cookies in the cookie store for the destination URL into the
+    request object, and then marks the ones that were not actual sent with the
+    request with the reason they were blocked.
+
+    It's not clear if we want to do the same, or just include the cookies that
+    were actually sent.
+
+1. Return a map matching the <code>NetworkRequestData</code>
+   production, with the <code>url</code> field set to |url|, the
+   <code>method</code> field set to |method|, the <code>headers</code> field set
+   to |headers|, the <code>headersSize</code> field set to |headers size| and
+   the <code>bodySize</code> field set to |body size|.
+
+#### The network.ResponseContent type #### {#type-network-ResponseContent}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+NetworkResponseContent = {
+    size: unit / null
+};
+</pre>
+
+The <code>ResponseContent</code> type represents the decoded response to a
+network request.
+
+<!-- Not sure this is worthwhile if it only ends up with a single field, but it
+would be natural to add a field here if we have a way to return the body -->
+
+#### The network.ResponseData type #### {#type-network-ResponseData}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+NetworkResponseData = {
+    url: text,
+    protocol: text,
+    status: uint,
+    statusText: text,
+    headers: [ *NetworkHeader ],
+    cookies: [ *NetworkCookie ],
+    mimeType: text,
+    headersSize: uint / null,
+    bodySize: uint / null,
+    bytesReceived: uint,
```

Comment:
Can this be different from `headersSize` + `bodySize`? A difference in the case of compression?